### PR TITLE
[webview_flutter] add debuggingEnabled property

### PR DESCRIPTION
**Pull request by lesnitsky**
## Description

Provides new property of `WebView` – debuggingEnabled (android only)

## Related Issues

flutter/flutter#29979

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide])
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java [hunk 1]
```diff
@@ -192,6 +192,10 @@ private void applySettings(Map<String, Object> settings) {
 
           webView.setWebViewClient(webViewClient);
           break;
+        case "debuggingEnabled":
+          final boolean debuggingEnabled = (boolean) settings.get(key);
+
+          webView.setWebContentsDebuggingEnabled(debuggingEnabled);
```
Comment (amirh) [inline packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: Is there an equivalent on iOS?

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 2]
```diff
@@ -205,6 +206,11 @@ class WebView extends StatefulWidget {
   /// [WebViewController.evaluateJavascript] can assume this.
   final PageFinishedCallback onPageFinished;
 
+  /// Allows remote WebView debugging if set to true
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: This Dart doc doesn't add much information.

Some prompts:
  * What's remote webview debugging?
  * Is it supported on Android and iOS? does it work the same on both platforms?

File: packages/webview_flutter/pubspec.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: webview_flutter
 description: A Flutter plugin that provides a WebView widget on Android and iOS.
-version: 0.3.5+2
+version: 0.3.5+3
```
Comment (amirh) [inline packages/webview_flutter/pubspec.yaml]: As this is adding a new feature Y in 0.X.Y+Z should be bumped, please also update the version in the changelog

**Comment by lesnitsky** [comment]
> Thanks for the contribution!
> 
> Left a few comments, other than that please add tests toggling the setting both ways.
> Ideally e2e tests, but I'm not sure if there's a reasonable way to verify that debugging was enabled, if there isn't a unit test is the best we can do.

e2e sounds overcomplicated for this usecase, unit test sounds good though
Thanks for other comments, absolutely valid points 👍

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java (hunk 1 above)
Comment (lesnitsky) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: debugging is enabled by default in dev build on iOS

**Comment by lesnitsky** [comment]
@amirh I've fixed all issues and added unit tests

File: packages/webview_flutter/lib/webview_flutter.dart (hunk 2 above)
Comment (lesnitsky) [reply packages/webview_flutter/lib/webview_flutter.dart]: updated

File: packages/webview_flutter/pubspec.yaml (hunk 3 above)
Comment (lesnitsky) [reply packages/webview_flutter/pubspec.yaml]: fixed

**Comment by lesnitsky** [comment]
@amirh fixed conflicts
could you please review this again after fixes to previous review?

**Comment by lesnitsky** [comment]
@bparrishMines I've noticed you were active on PRs recently. Could you let me know if I can do anything else to get this PR merged? Thanks

**Comment by amirh** [comment]
Thanks for following up, I merged master added a test and cleaned up documentation a bit.
Will land on green.

**Pull request merged**